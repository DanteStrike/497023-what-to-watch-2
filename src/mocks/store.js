import StoreNameSpace from "../reducers/store-name-space.js";
import {comments} from "./comments.js";
import {normalizedFilmsData} from "./films";

export const initStore = {
  [StoreNameSpace.APP]: {
    isReady: false,
    videoPlayerFilmID: -1
  },

  [StoreNameSpace.FILMS]: {
    data: {
      allIDs: [],
      byIDs: {}
    },

    promo: {
      filmID: null
    }
  },

  [StoreNameSpace.COMMENTS]: {
    postCommentState: {
      isSuccess: false,
      error: {
        isError: false,
        msg: ``
      }
    },
    curFilmComments: []
  },

  [StoreNameSpace.GENRES]: {
    currentFilter: `All genre`,
    data: {
      genres: [`All genre`],
      byGenres: {
        "All genre": [],
      }
    }
  },

  [StoreNameSpace.CATALOG]: {
    displayedFilmsAmount: 0
  },

  [StoreNameSpace.USER]: {
    auth: {
      isAuth: false,
      error: {
        isError: false,
        target: ``,
        msg: ``
      }
    },

    toggleFavoriteStatus: {
      isSuccess: false,
      error: {
        isError: false
      }
    },

    data: {
      id: null,
      email: ``,
      name: ``,
      avatarUrl: ``,
      myListFilmsIDs: []
    }
  }
};

export const loadedStore = {
  [StoreNameSpace.APP]: {
    isReady: true,
    videoPlayerFilmID: 1
  },

  [StoreNameSpace.FILMS]: {
    data: normalizedFilmsData,

    promo: {
      filmID: 3
    }
  },

  [StoreNameSpace.COMMENTS]: {
    postCommentState: {
      isSuccess: true,
      error: {
        isError: false,
        msg: ``
      }
    },
    curFilmComments: comments
  },

  [StoreNameSpace.GENRES]: {
    currentFilter: `All genre`,
    data: {
      genres: [`All genre`, `genreOne`, `genreTwo`],
      byGenres: {
        "All genre": [1, 3],
        "genreOne": [1],
        "genreTwo": [3]
      }
    }
  },

  [StoreNameSpace.CATALOG]: {
    displayedFilmsAmount: 0
  },

  [StoreNameSpace.USER]: {
    auth: {
      isAuth: true,
      error: {
        isError: true,
        target: `target`,
        msg: `any`
      }
    },

    toggleFavoriteStatus: {
      isSuccess: true,
      error: {
        isError: false
      }
    },

    data: {
      id: 1,
      email: `Oliver.conner@gmail.com`,
      name: `Oliver.conner`,
      avatarUrl: `img/1.png`,
      myListFilmsIDs: [3]
    }
  },
};
