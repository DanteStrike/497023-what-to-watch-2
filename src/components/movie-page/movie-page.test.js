import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from "react-router-dom";
import MoviePage from "./movie-page.jsx";

import {loadedStore} from "../../mocks/store.js";
import {createStore} from "redux";
import {Provider} from "react-redux";


it(`Render correctly MoviePage component`, () => {
  const store = createStore(() => loadedStore);
  const component = renderer
    .create(
        <Provider store={store}>
          <Router>
            <MoviePage
              curFilmID={1}
            />
          </Router>
        </Provider>
    ).toJSON();

  expect(component).toMatchSnapshot();
});
