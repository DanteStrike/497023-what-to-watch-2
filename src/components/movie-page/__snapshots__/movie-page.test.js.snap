// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly MoviePage component 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="filmOne"
          src="img/filmOne.jpg"
          style={
            Object {
              "backgroundColor": "#ffffff",
            }
          }
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link "
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            filmOne
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              genreOne
            </span>
            <span
              className="movie-card__year"
            >
              2019
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/films/1/add-review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="filmOne"
            height="327"
            src="img/filmOne-poster.jpg"
            width="218"
          />
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div>
        MOVIE LIST
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
